### PR TITLE
austin.intermediate.summer2.calderoni.giovanna

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: apps/02RockPaperScissors.js [hunk 1]
```diff
@@ -4,13 +4,47 @@ var assert = require('assert');
 var prompt = require('prompt');
 prompt.start();
 
-
+var hand1Wins = "Hand one wins!";
```
Comment (tjefferson08) [inline apps/02RockPaperScissors.js:7]: Sweeeeeeet

File: apps/02RockPaperScissors.js [hunk 2]
```diff
@@ -4,13 +4,47 @@ var assert = require('assert');
...
+var hand2Wins = "Hand two wins!"
 function rockPaperScissors(hand1, hand2) {
 
     // Write code here
+    // If this statement is true, the code reader quits reading the code.
+    if (hand1 === hand2) {
+      return "It's a tie!"
+    }
+
+    if (hand1 === "rock") {
```
Comment (tjefferson08) [inline apps/02RockPaperScissors.js:17]: This is a little verbose, but it shows that you definitely understand the logic. I like it! This is how I wrote it too :D

File: apps/02RockPaperScissors.js [hunk 3]
```diff
@@ -4,13 +4,47 @@ var assert = require('assert');
...
+      }
+      if (hand1 === "paper") {
+        if (hand2 === "rock") {
+          return hand1Wins
+        }
+        else if (hand2 === "scissors") {
+          return hand2Wins
+        }
+      }
+    // Here you need to write a statement in case the user doesn't enter a valid response.
```
Comment (tjefferson08) [inline apps/02RockPaperScissors.js:41]: Nice note to your future self!

File: homework/01LessonOne.js [hunk 4]
```diff
@@ -16,28 +16,28 @@ var myUndefined;
 // Let's create a new variable named myNull and give it the value null.
 //
 // What is the difference between null and undefined?
-
-var myNull;
+        // Null means nothing has been declared and undefined has been declared but has not been assigned a value.
```
Comment (tjefferson08) [inline homework/01LessonOne.js]: > Null means nothing has been declared 

Not exactly! Think of `null` like a placeholder value

File: homework/01LessonOne.js [hunk 5]
```diff
@@ -46,57 +46,57 @@ var myString;
...
 // What is the value of true && false
 
-var trueAndFalse;
+var trueAndFalse = false;
 
 // Problem 9:
 // What is the value of false && false
 
-var falseAndFalse;
+var falseAndFalse = false;
```
Comment (tjefferson08) [inline homework/01LessonOne.js:63]: FUN!

File: homework/01LessonOne.js [hunk 6]
```diff
@@ -107,34 +107,33 @@ var notNotTrue;
 //
 // Your Answer Goes Here:
 //
-// ****
-
+// **** When you use &&  you want know if your conditional results are true or false. When you use || you want all your conditonal results to be true.
```
Comment (tjefferson08) [inline homework/01LessonOne.js]: Whoops! This is a little off. Let's talk about it! And read here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators

File: homework/01LessonOne.js [hunk 7]
```diff
@@ -145,55 +144,56 @@ var notNotUndefined;
 //
 // Your Answer Goes Here:
 //
-// ****
-
+// **** truthiness is when a value has been assigned and falsiness is when a value has not been assigned.
```
Comment (tjefferson08) [inline homework/01LessonOne.js]: Let's talk a bit more about this! Truthy/falsy isn't quite the same as assigned/undefined

File: homework/01LessonOne.js [hunk 8]
```diff
@@ -203,14 +203,17 @@ var stringFourStrictNotEqualsNumberFour;
 // What’s the difference between == and ===?
 //
 // Your Answer Goes Here:
-//
-//
+//          == returns true if both operators are equal.
+//           === is a strict equal and returns true if both are equal and of the same type.
```
Comment (tjefferson08) [inline homework/01LessonOne.js]: Simple, I like it

File: homework/01LessonOne.js [hunk 9]
```diff
@@ -203,14 +203,17 @@ var stringFourStrictNotEqualsNumberFour;
...
 // What’s the difference between != and !==?
 //
 // Your Answer Goes Here:
 //
 // ****
-
+//            != determines the "truth" of what a value is not
```
Comment (tjefferson08) [inline homework/01LessonOne.js]: This is a little trickier than it sounds, let's catch up on the details

File: homework/02LessonTwo.js [hunk 10]
```diff
@@ -26,6 +26,12 @@ var alwaysTrue = function () {
 // argument1 is equal to argument2
 
 function equals(argument1, argument2) {
+  if (argument1 == argument2) {
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:29]: boom!

File: homework/02LessonTwo.js [hunk 11]
```diff
@@ -34,17 +40,33 @@ function equals(argument1, argument2) {
...
+  }
+  else{
+    return false;
+  }
 }
 
 // Problem 3:
 // write a function named add(number1, number2)
 // add will add two numbers and return the result
+    var number1 = 2;
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:54]: We don't need these variables, but they aren't hurting anything

File: homework/02LessonTwo.js [hunk 12]
```diff
@@ -34,17 +40,33 @@ function equals(argument1, argument2) {
...
 // Problem 4:
 // write a function named addThree(number1, number2, number3)
 // this function will add three numbers
 // you must call your function add() in addThree()
-
+        var number1 = 90;
+        var number2 = 45;
+        var number3 = 10000;
+        function addThree(number1, number2, number3){
+          return number1 + number2 + number3;
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:68]: Note! Gotta use the add() function from up above. no plus signs allowed

File: homework/02LessonTwo.js [hunk 13]
```diff
@@ -54,21 +76,24 @@ function lessThanOrEqualTo(parameter1, parameter2) {
 // What is a function? How do you define a function in Javascript?
 //
 // Your Answer Goes Here:
-//
+//    a function is a way to solve problems easily
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:79]: MOAR DEETS 20 word minimum :)

File: homework/02LessonTwo.js [hunk 14]
```diff
@@ -54,21 +76,24 @@ function lessThanOrEqualTo(parameter1, parameter2) {
...
 //
 // What is a return value?
 //
 // Your Answer Goes Here:
-//
+//    a return value is what you would like your function to turn out as
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:84]: I'll take it

File: homework/02LessonTwo.js [hunk 15]
```diff
@@ -54,21 +76,24 @@ function lessThanOrEqualTo(parameter1, parameter2) {
...
 //
 // How do you define a named function?
 //
 // Your Answer Goes Here:
-//
+//    a named  function help us tackle large problems, making them easier to solve.
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:89]: Can you give us a syntax example here?

File: apps/03TicTacToe.js [hunk 16]
```diff
@@ -22,24 +22,74 @@ function printBoard() {
...
+    return true;
+  }
+  if (board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn) {
+    return true;
+  }
+  if (board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn) {
+  return true;
+  }
+  else {
+      return false;
```
Comment (tjefferson08) [inline apps/03TicTacToe.js:35]: This is legit, but it looks like you were thinking of an if/else if/else structure. 

Tying the else to just the last `if` works in this case, but it isn't perfect logic (always striving for perfection :D)

File: apps/03TicTacToe.js [hunk 17]
```diff
@@ -22,24 +22,74 @@ function printBoard() {
...
+  }
+};
+
 function ticTacToe(row, column) {
-    // Your code here
-}
+  board [row][column] = playerTurn;
```
Comment (tjefferson08) [inline apps/03TicTacToe.js:87]: Noice

File: homework/03LessonThree.js [hunk 18]
```diff
@@ -20,16 +20,15 @@ var firstName = 'susan';
...
 // Problem 2:
 // What is the value of lastName at this point in the file?
-var valueOfLastName;
+var valueOfLastName = 'plaid';
 
 // Problem 3:
 // Fix the following function so that it returns the first argument
+var firstArgument = 'bob';
 function returnTheFirstArgument(firstArgument) {
-  var firstArgument = 'bob';
```
Comment (tjefferson08) [inline homework/03LessonThree.js:32]: Sweet

File: homework/03LessonThree.js [hunk 19]
```diff
@@ -41,7 +40,7 @@ function returnTheFirstArgument(firstArgument) {
 // What is scope?
 //
 // Your Answer Goes Here:
-//
+// It's right before the function and inside the function.
```
Comment (tjefferson08) [inline homework/03LessonThree.js:43]: I like this definition, but I'd say it's EVERYTHING before the function, not just _right_ inside

File: homework/03LessonThree.js [hunk 20]
```diff
@@ -89,17 +88,18 @@ var problemSevenAnswer;
...
 // Create an array with 5 elements in it
-var lengthFiveArray;
+var lengthFiveArray = [ 1, 2, 3, 4, 5 ];
 
 // DO NOT MODIFY
 var nameArray = ['bob', 'fred', 'susan'];
 
 // Problem 10:
 // Replace the value 'fred' in nameArray with 'george'
+    var nameArray = [ 'bob', 'george', 'susan' ]
```
Comment (tjefferson08) [inline homework/03LessonThree.js:102]: Technically right, but we want you to use the indexing syntax to change the original `nameArray`.
So try this one again without redeclaring nameArray

File: apps/05Mastermind.js [hunk 21]
```diff
@@ -26,14 +28,51 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
+function generateHint(solution, guess) {
     // your code here
+    var solutionArray = guess.split("");
+    var guessArray = solution.split("");
```
Comment (pieroapretto) [inline apps/05Mastermind.js]: You have the two parameters assigned out of order here.

Comment (ghost) [reply apps/05Mastermind.js]: That's too funny! Thank you for your help!! 💯

File: homework/01LessonOne.js [hunk 22]
```diff
@@ -9,34 +9,37 @@ var assert = require('assert');
 // Problem 0:
```
Comment (ghost) [inline homework/01LessonOne.js:9]: Hey there... My bad. I realized I never finished this Here you go & if it's too late that's okay, still wanted to finish it.

File: apps/04TowersOfHanoi.js [hunk 23]
```diff
@@ -16,24 +16,41 @@ function printStacks() {
     console.log("c: " + stacks.c);
 }
```
Comment (ghost) [inline apps/04TowersOfHanoi.js:18]: So IDK what happened but my original code before commit-ing doesn't look like this... I erased all of the `//Your code here` lines so my code was flush and definitely didn't repeat any unneccesary lines! lol

File: apps/05Mastermind.js [hunk 24]
```diff
@@ -13,7 +13,7 @@ var index;
 
 function printBoard() {
     for (index = 0; index < board.length; index++) {
-        console.log(board[i])
```
Comment (ghost) [inline apps/05Mastermind.js:14]: Umm... IDK where this line came from... It's not on my pre-commit code.

File: apps/05Mastermind.js [hunk 25]
```diff
@@ -30,8 +30,8 @@ function getRandomInt(min, max) {
 
 function generateHint(solution, guess) {
     // your code here
-    var solutionArray = guess.split("");
```
Comment (ghost) [inline apps/05Mastermind.js]: I know lines 33 & 34 are wrong but I fixed them when I resubmitted them, so I don't get why they are still showing up... IDK why Codacy doesn't like me!!

File: apps/05Mastermind.js [hunk 26]
```diff
@@ -42,8 +42,8 @@ function generateHint(solution, guess) {
       }
     }
 
-    for (index=0; index < guessArray.length; index++) {
```
Comment (ghost) [inline apps/05Mastermind.js]: And... Again, the red lines 45 & 46 are also not in my pre-commit code. So! What am I doing wrong?? I don't like seeing red X's by my submissions...

File: apps/05Mastermind.js [hunk 27]
```diff
@@ -26,14 +28,52 @@ function getRandomInt(min, max) {
...
+        correctLetterLocations++;
+        solutionArray[index] = null;
+      }
+    }
+
+    for (index= 0; index < guessArray.length; index++) {
+    var targetIndex = solutionArray.indexOf(guessArray[index]);
+    if (guessArray.indexOf(solutionArray[index]) > -1) {
+      correctLetters++;
+      solutionArray[index] = null;
```
Comment (tjefferson08) [inline apps/05Mastermind.js:49]: Hmmm something about this line looks a little off to me. Shouldn't we be using the `targetIndex` variable?

File: homework/06LessonSix.js [hunk 28]
```diff
@@ -20,22 +22,27 @@ var point = {
 // In order to do this, you should add parameters x and y
 // as properties to "this".
 // example: this.propName = propValue;
-function Point(x, y) {
-}
+function Point(paramX, paramY) {
```
Comment (tjefferson08) [inline homework/06LessonSix.js:25]: Nice. I like differently-named params

File: homework/06LessonSix.js [hunk 29]
```diff
@@ -20,22 +22,27 @@ var point = {
...
+  this.x = paramX;
+  this.y = paramY;
+};
 
 // Problem 3:
 // Create a new point using the class constructor Point.
 // Set x to 5, and y to -3
 // This can be done as follows: new Point(someX, someY)
-var anotherPoint;
+var anotherPoint = new Point( );
```
Comment (tjefferson08) [inline homework/06LessonSix.js:34]: We can improve this. `Point(paramX, paramY)` has params that let us automatically set the x and y properties. Let's use them!

File: homework/06LessonSix.js [hunk 30]
```diff
@@ -96,7 +108,11 @@ var conferenceRoom = new ConferenceRoom();
 // Write your answer in comments
 //
 // What is a method?
-//
+//  Method is a type of function that has "special" attributes.
```
Comment (tjefferson08) [inline homework/06LessonSix.js:111]: Nice

File: apps/06SpaceTravelToMars.js [hunk 31]
```diff
@@ -9,7 +9,37 @@ var jobTypes = {
...
+  this.job = job;
+  this.specialSkill = specialSkill;
+  //this.ship = null;
+  this.enterShip = function(ship){
+    this.ship = ship;
+    ship.crew.push(this)
+  }
+}
+
+var rick = new CrewMember('Rick Martinez', 'pilot', 'chemistry');
```
Comment (tjefferson08) [inline apps/06SpaceTravelToMars.js:23]: Don't really need this guy, but it's all good. The code looks great overall.

File: apps/07jquery-tic-tac-toe/script.js [hunk 32]
```diff
@@ -1,5 +1,71 @@
...
+
+};
+
+function checkForWin() {
+
+if (horizontalWin() || verticalWin() || diagonalWin()){
+  $("#announce-winner").text('player ' + playerTurn + ' wins!');
+  return true;
+}
+};
```
Comment (tjefferson08) [inline apps/07jquery-tic-tac-toe/script.js:70]: Perfect!

File: apps/06SpaceTravelToMars.js (hunk 31 above)
Comment (ghost) [reply apps/06SpaceTravelToMars.js:23]: Well originally my thinking was, whoosh there are a bunch of crewMembers, I'm going to need to create variables for ALL of them!! So I wrote statements for them using celebrities: Matthew McCaughey, Oprah, Obama, and Kevin James... It was going to be 1 mad crazy interesting trip to Mars! lol

**Pull request closed without merging**